src/parser/ion_1_0/text/time.rs

```diff
@@ -0,0 +1,164 @@
+                second,
+                // TODO: leading zeroes need to be added.
+                fraction_coefficient.to_str_radix(10)
+            )
+            .fmt(f),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct TextTimestamp {
+    date: TextDate,
+    time: Option<TextTime>,
+    offset: UtcOffset,
+}
+
+impl TextTimestamp {
+    pub fn new(date: TextDate, time: Option<TextTime>, offset: time::UtcOffset) -> Self {
+        Self { date, time, offset }
+    }
+}
+
+impl TryFrom<TextTimestamp> for Timestamp {
+    type Error = TextFormatError;
+
+    fn try_from(timestamp: TextTimestamp) -> Result<Self, Self::Error> {
+        let offset = timestamp.offset.as_seconds();
+        Ok(match timestamp.time {
+            None => match timestamp.date {
+                TextDate::Year { year } => Timestamp::Year { year, offset },
+                TextDate::Month { year, month } => Timestamp::Month {
+                    year,
+                    month,
+                    offset,
+                },
+                TextDate::Day { date } => Timestamp::Day {
+                    year: date.year() as u16,
+                    month: date.month(),
+                    day: date.day(),
+                    offset,
+                },
+            },
+            Some(time) => match timestamp.date {
+                TextDate::Day { date } => match time {
+                    TextTime::Minute { hour, minute } => Timestamp::Minute {
+                        year: date.year() as u16,
+                        month: date.month(),
+                        day: date.day(),
+                        hour,
+                        minute,
+                        offset,
+                    },
+                    TextTime::Second {
+                        hour,
+                        minute,
+                        second,
+                    } => Timestamp::Second {
+                        year: date.year() as u16,
+                        month: date.month(),
+                        day: date.day(),
+                        hour,
+                        minute,
+                        second,
+                        offset,
+                    },
+                    TextTime::FractionalSecond {
+                        hour,
+                        minute,
+                        second,
+                        fraction_coefficient,
+                        fraction_exponent,
+                    } => Timestamp::FractionalSecond {
+                        year: date.year() as u16,
+                        month: date.month(),
+                        day: date.day(),
+                        hour,
+                        minute,
+                        second,
+                        fraction_coefficient,
+                        fraction_exponent,
+                        offset,
+                    },
+                },
+                _ => return Err(TextFormatError::ImpreciseDate),
```

Comment:
What is not getting covered in this match?

Reply:
`TextDate::Year` and `TextDate::Month`. This is a little awkward as you noted in #12 with the split date/time types since there's no guarantee that `TextDate` has full precision when we find a time (based on the types alone).